src/discord/commands/book.ts

```diff
@@ -181,27 +185,81 @@ export default {
 
         const selectMenuCollector = message.createMessageComponentCollector({ componentType: 'SELECT_MENU', time: 120000 });
 
         selectMenuCollector.on('collect', async (menuInteraction: SelectMenuInteraction) => {
             switch (menuInteraction.customId) {
-                case 'dateSelectMenu':
+                case 'dateSelectMenu': {
+                    menuSelectedDate = menuInteraction.values[0];
                     selectMenuDate = new MessageActionRow().addComponents(
-                        new MessageSelectMenu().setCustomId('dateSelectMenu').setPlaceholder('Select Date of Room Booking').addOptions(getDateOptions(menuInteraction.values[0]))
+                        new MessageSelectMenu().setCustomId('dateSelectMenu').setPlaceholder('Select Date of Room Booking').addOptions(getDateOptions(menuSelectedDate))
                     );
-                    avaliableTimeblocks = new MessageActionRow().addComponents(parseTimeblocks((await searchTimeblocks(menuInteraction.values[0], sectionInformation!, roomInformation!))!));
+                    const avaliableTimeblocks = new MessageActionRow().addComponents(parseTimeblocks(await searchTimeblocks(menuSelectedDate, sectionInformation!, roomInformation!)));
 
                     menuInteraction.update({ components: [selectMenuDate, avaliableTimeblocks] });
                     break;
-                case 'timeBlockSelectMenu':
-                    //Temporary method for booking a room (postponed until database structure + user integration is fleshed out)
-                    await menuInteraction.reply(`Timeslot ${menuInteraction.values[0]} supposedly booked!`);
+                }
+                case 'timeBlockSelectMenu': {
+                    //ESLint disabled for next line as regex is correct at removing unicode characters. Removes hidden unicode U+200E character that invalidates parseInt()/toJSDate()
+                    const selectedTimeblock = menuInteraction.values[0].split(',').map((element: string) => element.replace(/[^\x00-\x7F]/g, '')); //eslint-disable-line
+
+                    const selectedDate = DateTime.fromFormat(menuSelectedDate, 'yyyy-MM-dd', { zone: 'America/Toronto' });
+                    const _startsAt = selectedDate.set({ hour: parseInt(selectedTimeblock[0]) }).toJSDate();
+                    const _endsAt = selectedDate.set({ hour: parseInt(selectedTimeblock[1]) }).toJSDate();
+                    const maxCapacity = parseInt(selectedTimeblock[2]);
+
+                    const foundTimeblock = await TimeblockModel.findOne({ sectionId: Types.ObjectId(_sectionId), startsAt: _startsAt, endsAt: _endsAt });
+                    let bookingId;
+
+                    if (foundTimeblock) {
+                        const potentialTimeblock = await BookingModel.find({ timeBlock: foundTimeblock!._id, booker: interaction.user.id });
```

Comment:
Should we be using `findOne()` here? I can't remember what the behaviour of .find is when nothing exists but it might just be an empty array which is truthy.